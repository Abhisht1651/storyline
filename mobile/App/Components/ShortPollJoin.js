// @flow

import React from 'react'
import { View, Text, Image } from 'react-native'

// Styles
import styles from './Styles/ShortPollJoinStyle'
import images from '../Themes/Images'

type Props = {
  border: boolean,
  question: string,
  votes: number,
  timeLeft: number,
  isPollActive: boolean,
  voterIcons: Array<string>
}

const shortPollJoin = (props: Props) => {
  const { border, question, votes, timeLeft, isPollActive, voterIcons } = props
  const { pollActiveIcon, pollInactiveIcon } = images
  const pollIcon = isPollActive ? pollActiveIcon : pollInactiveIcon
  const pollText = isPollActive ? `${timeLeft} hours left` : 'Poll ended'

  const voterIconsElement = voterIcons.map((voterIcon, index) => (
    <Image source={voterIcon} key={index} style={styles.voterIcon} />
  ))

  return (
    <View style={border ? styles.container : styles.containerWithoutBorder}>
      <View style={styles.joinTextContainer}>
        <Text style={styles.joinText}>Join to this poll!</Text>
      </View>
      <View style={styles.divider} />
      <View style={styles.timeContainer}>
        <View style={styles.timeIconContainer}>
          <Image source={pollIcon} style={styles.pollIcon} />
        </View>
        <View style={styles.timeTextContainer}>
          <Text style={styles.timeText}>{pollText}</Text>
        </View>
      </View>
      <View style={styles.questionContainer}>
        <Text style={styles.questionText}>{question}</Text>
      </View>
      <View style={styles.voterContainer}>
        <View style={styles.voterIconContainer}>{voterIconsElement}</View>
        <View style={styles.votesContainer}>
          <Text style={styles.votes}>{votes}K have voted</Text>
        </View>
      </View>
    </View>
  )
}

shortPollJoin.defaultProps = {
  border: true,
  question: 'Who are your favorites for the World Cup 2020?',
  votes: 54,
  timeLeft: 2,
  isPollActive: true,
  voterIcons: [images.voterIcon1, images.voterIcon2, images.voterIcon3]
}

export default shortPollJoin
