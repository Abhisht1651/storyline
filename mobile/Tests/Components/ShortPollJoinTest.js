// @flow
import 'react-native'
import React from 'react'
import { shallow } from 'enzyme'
import ShortPollJoin from '../../App/Components/ShortPollJoin'

describe('ShortPollJoin', () => {
  it('ShortPollJoin component renders correctly', () => {
    const wrapper = shallow(<ShortPollJoin />)
    expect(wrapper).toMatchSnapshot()
  })
  it('should render the Text Component', () => {
    const wrapper = shallow(<ShortPollJoin />)
    expect(wrapper.find('Text')).toHaveLength(4)
  })
  it('should render the Image Component', () => {
    const wrapper = shallow(<ShortPollJoin />)
    expect(wrapper.find('Image')).toHaveLength(4)
  })
})
