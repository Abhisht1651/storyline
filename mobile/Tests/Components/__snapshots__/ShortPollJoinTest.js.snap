// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShortPollJoin ShortPollJoin component renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FFFFFF",
      "borderColor": "#E6ECED",
      "borderRadius": 30,
      "borderStyle": "solid",
      "borderWidth": 1,
      "minHeight": 206,
      "minWidth": 250,
      "opacity": 1,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "paddingVertical": 17,
        "width": "83.6%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#1F1F60",
          "fontFamily": "BananaGrotesk-Regular",
          "fontSize": 32,
          "letterSpacing": -0.11,
          "lineHeight": 44,
          "opacity": 1,
          "textAlign": "left",
        }
      }
    >
      Join to this poll!
    </Text>
  </View>
  <View
    style={
      Object {
        "borderBottomColor": "#E6ECED",
        "borderBottomWidth": 1,
        "width": "100%",
      }
    }
  />
  <View
    style={
      Object {
        "flexDirection": "row",
        "paddingTop": 35,
        "width": "83.6%",
      }
    }
  >
    <View
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Image
        source={
          Object {
            "default": "",
          }
        }
        style={
          Object {
            "height": 34,
            "width": 34,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "paddingHorizontal": 5,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#002468",
            "fontFamily": "BananaGrotesk-Regular",
            "fontSize": 28,
            "letterSpacing": -0.07,
            "opacity": 0.5,
            "textAlign": "left",
          }
        }
      >
        2 hours left
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "marginTop": 13.5,
        "width": "83.6%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#1F1F60",
          "fontFamily": "BananaGrotesk-Medium",
          "fontSize": 36,
          "letterSpacing": -0.12,
          "lineHeight": 44,
          "opacity": 1,
          "textAlign": "left",
        }
      }
    >
      Who are your favorites for the World Cup 2020?
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "marginBottom": 28,
        "marginTop": 9,
        "width": "83.6%",
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "marginLeft": 16,
        }
      }
    >
      <Image
        key="0"
        source={
          Object {
            "default": "",
          }
        }
        style={
          Object {
            "borderRadius": 28,
            "height": 56,
            "marginLeft": -16,
            "opacity": 1,
            "width": 56,
          }
        }
      />
      <Image
        key="1"
        source={
          Object {
            "default": "",
          }
        }
        style={
          Object {
            "borderRadius": 28,
            "height": 56,
            "marginLeft": -16,
            "opacity": 1,
            "width": 56,
          }
        }
      />
      <Image
        key="2"
        source={
          Object {
            "default": "",
          }
        }
        style={
          Object {
            "borderRadius": 28,
            "height": 56,
            "marginLeft": -16,
            "opacity": 1,
            "width": 56,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "marginLeft": 10,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#082366",
            "fontFamily": "BananaGrotesk-Regular",
            "fontSize": 28,
            "letterSpacing": -0.1,
            "opacity": 0.5,
            "textAlign": "left",
          }
        }
      >
        54
        K have voted
      </Text>
    </View>
  </View>
</View>
`;
